invoices/invoice_expiry_watcher.go

```diff
@@ -174,15 +207,17 @@ func (ew *InvoiceExpiryWatcher) mainLoop() {
 
 	for {
 		// Cancel any invoices that may have expired.
-		ew.cancelExpiredInvoices()
+		ew.cancelNextExpiredInvoice()
```

Comment:
Hmm, unsure if this is better than the prior diff. On one hand we'll always check for new invoices before consuming the existing set (partially addressing our starvation issue), but on the other-hand we'll now execute the main loop multiples time to chip away at the set of expired invoices.

Reply:
One option to counter act this is to buffer the `newInvoices` channel to a degree.

Reply:
Imo it's better as whoever called `InvoiceRegistry.AddInvoice` may still be blocked until we cancel everyting as we won't read from the `newInvoices` channel.

Reply:
> Imo it's better as whoever called InvoiceRegistry.AddInvoice may still be blocked until we cancel everyting as we won't read from the newInvoices channel.

Good point, but in the current form we still can't guarantee that reading from `newInvoices` takes precedence over `nextExpiry`, so I think we'd want to add a non-blocking select statement before the current one:

```go
select {
case <-ew.newInvoices:
default:
}
```

Doing this will likely cause even higher CPU usage however.